### PR TITLE
Fix #8072: Portfolio v2

### LINKED ISSUE #8072: Portfolio v2
### Description: 

Integrate Portfolio v2 designs with NFT tab integrated into Portfolio.

https://www.figma.com/file/muVcmcu4RFNu0LkXe9VRDl/Wallet-2.0?type=design&node-id=4738-346151&mode=design&t=NBz1FjJKMDbkYMY8-4

### PR DESCRIPTION
**Pull request by StephenHeaps**
## Summary of Changes
- Merge NFT tab into Portfolio tab, and remove existing NFT tab from top page control
- Update Portfolio to use new v2 designs from Figma.
- Separated most views / content views outside of `PortfolioView`, header, segmented control, assets list, etc.

This pull request fixes #8072

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:
1. Open Portfolio, verify Buy / Send / Swap buttons are all opening their expected views
2. Tap the graph timeframe selector, verify graph updates when changed.
    - Price indicator should be green for balance increases, red for balance decreases (from start of graph timeframe)
3. Tap on the Assets / NFTs selector, verify displayed assets update to tapped on page as expected
4. Tap on individual assets & NFTs, verify their detail views open as expected
5. Switch to Assets section and open filters. Change filters to group by network or account & verify groups are outlined with a line to distinguish the groups when expanded and collapsed.

## Screenshots:
Portfolio v2 - light | Portfolio assets v2 - light | Portfolio nfts v2 - light
--|--|--
Portfolio v2 - dark | Portfolio assets v2 - dark | Portfolio nfts v2 - dark

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Sources/BraveWallet/Crypto/Stores/PortfolioStore.swift [hunk 1]
```diff
@@ -186,10 +186,18 @@ struct BalanceTimePrice: DataPoint, Equatable {
...
+  let priceDifference: String
+  let percentageChange: String
+  let isBalanceUp: Bool
+}
+
 /// A store containing data around the users assets
 public class PortfolioStore: ObservableObject, WalletObserverStore {
   /// The dollar amount of your portfolio
   @Published private(set) var balance: String = "$0.00"
+  ///
```
Comment (nuo-xu) [inline Sources/BraveWallet/Crypto/Stores/PortfolioStore.swift]: forgot to add the comment?

File: Sources/BraveWallet/Crypto/Stores/PortfolioStore.swift [hunk 2]
```diff
@@ -502,6 +510,26 @@ public class PortfolioStore: ObservableObject, WalletObserverStore {
...
       }
+      
+      if let oldestHistoricalValue = historicalBalances.first {
+        let priceDifference = currentBalance - oldestHistoricalValue.price
+        let percentageChange = priceDifference / oldestHistoricalValue.price * 100
+        let isBalanceUp = priceDifference > 0
+        balanceDifference = .init(
+          priceDifference: String(
+            format: "%@%@",
+            isBalanceUp ? "+" : "", // include plus if balance increased
```
Comment (nuo-xu) [inline Sources/BraveWallet/Crypto/Stores/PortfolioStore.swift:521]: there is no "-" for down?

**Review by nuo-xu — APPROVED** [review]
Screenshot 2023-10-05 at 4 23 33 PM
can you double check the icon padding. it doesnt look like centred.

File: Sources/BraveWallet/Crypto/Stores/PortfolioStore.swift (hunk 2 above)
Comment (StephenHeaps) [reply Sources/BraveWallet/Crypto/Stores/PortfolioStore.swift:521]: The `-` gets added automatically for negative numbers 🙂
down

File: Sources/BraveWallet/Crypto/Stores/PortfolioStore.swift (hunk 1 above)
Comment (StephenHeaps) [reply Sources/BraveWallet/Crypto/Stores/PortfolioStore.swift]: Added comment in https://github.com/brave/brave-ios/pull/8140/commits/8ff2febc76b935a459487f20c4421f7a334d6125

**Review by nuo-xu — APPROVED** [review]

**Pull request merged**